### DISCUSSION TITLE
COMDIRAC profiles & dinit removal

### DISCUSSION THREAD

**Discussion by sfayer** [DIRACGrid/DIRAC#6869] [General]
Hi,

This is mainly a question for @atsareg initially...

As part of the COMDIRAC v8r1 cleanup we've been asked to remove dinit if possible... The current workflow for a COMDIRAC as I understand it is:
- User runs `dinit <profile_name>`, profile_name is usually the DIRAC group, but doesn't have to be.
- This then does the proxy init stuff.
- The session cache file is created based on the parent PID.
- Other COMDIRAC commands are run and find the same cache as they have the same parent PID.

Our suggested new workflow is:
- User runs `dirac-proxy-init -g <dirac_group>` to get a proxy, no immediate session initialisation happens.
- When any COMDIRAC command is run (or more specifically a new DSession() object is created), it checks if there is a session cache file using PPID as usual and if there isn't automatically initialises it.

We would probably also remove all references to DCOMMANDS_PPID environment variable as it isn't clear to me that anyone is using that now: It isn't set in any of the current bashrc/diracosrc init files any more, so all of the commands fall back to just auto-detecting the PPID.

This would mean that there would no longer be any custom profiles: You'd have one profile per DIRAC group that you use. Does that sound reasonable to you?

Regards,
Simon

**Comment by marianne013** [comment]
@IgorPelevanyuk and @adriansev  As you seem to be using COMDIRAC, do you have any opinion on this ?

**Reply by IgorPelevanyuk** [reply]
Hello Daniela and Simon! I do not have strong opinion on that topic. 
I personaly prefer dinit group_name. It is shorter.
About profiles, I just use the default that is done by dconfig -g. 
Generally, nothing will change for me (as far as I see it now). All these changes look ok for me.

**Comment by adriansev** [comment]
Hi! Not really, my use-case for dinit was just `dinit -p` so, if session is automatically created then it's fine .. but, you could also migrate the `-p` to `dirac-proxy-init` :)

**Comment by marianne013** [comment]
Implemented in: https://github.com/DIRACGrid/DIRAC/pull/6961 (will go in v8.1 onwards)
